### PR TITLE
fix: lint-staged for solidity

### LINKED ISSUE #447: Solidity Prettier Misconfiguration
image

Source: https://t.me/UbiquityDevPool/2303

> Lint-staged might be misconfigured,
I had use --no-verify a few times.

> We should ideally verify that it’s misconfigured and then fix it

> Although when I tried to make a commit today I didn’t encounter any issues again

> Perhaps it’s only for solidity files and not typescript files? I wouldn’t really know as I never touch the solidity code

> Yes. I agree. Prettier has an override for .sol files. I think that’s what is causing it

### PR DESCRIPTION
**Pull request by Steveantor**
- install `prettier-plugin-solidity` for prettier
- update `.prettierrc `
- update _lint-staged_ in `package.json`

Resolves #447

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,10 +39,14 @@
     "postinstall": "husky install"
   },
   "lint-staged": {
-    "*.{js,jsx,ts,tsx,json,sol}": [
+    "*.{js,jsx,ts,tsx,json}": [
       "prettier --write",
       "eslint --fix",
       "cspell --no-must-find-files"
+    ],
+    "*.sol": [
```
Comment (0x4007) [inline package.json:47]: Logically, this is the same. What's the difference?

File: .prettierrc [hunk 2]
```diff
@@ -14,9 +16,8 @@
         "tabWidth": 4,
         "useTabs": false,
         "singleQuote": false,
-        "bracketSpacing": false,
-        "explicitTypes": "always"
```
Comment (0x4007) [inline .prettierrc:18]: I guess this applies for everything but *.sol ?

Comment (Steveantor) [reply .prettierrc:18]: It only applies to *.sol,
_explicitTypes_ is unsupported option for solidity linter.

Comment (Steveantor) [reply .prettierrc:18]: > I guess this applies for everything but *.sol ?

It only applies to *.sol,
_explicitTypes_ is unsupported option for solidity linter. 
@pavlovcik

File: package.json (hunk 1 above)
Comment (Steveantor) [reply package.json:47]: > Logically, this is the same. What's the difference?

not running `eslint --fix` with *.sol files.
@pavlovcik

**Review by 0x4007 — APPROVED** [review]

**Pull request merged**